packages/remix-server-runtime/router/utils.ts

```diff
@@ -817,7 +817,8 @@ export function resolvePath(to: To, fromPathname = "/"): Path {
   } = typeof to === "string" ? parsePath(to) : to;
 
   let pathname = toPathname
-    ? toPathname.startsWith("/")
+    ? // we don't want to prepend the fromPathname on root or external toPathnames'
+      toPathname.startsWith("/") || toPathname.startsWith("http")
```

Comment:
@brophdawg11 do we want to accept other protocols? (app:// itunes:// ..)

Reply:
Yeah definitely.  I did this in RR the way it's checked in Remix today which should account for other protocols: https://github.com/remix-run/react-router/pull/9590/files#diff-c6f085a772081501e6db2af3eee90b15fe7cd7c965d2f220c2a3c5c7772bc0c4R2568